src/utils/useRoutedTabs.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react'
+import { useHistory, useLocation } from 'react-router-dom'
+
+/**
+ * Hook to get and set the current tab, while updating the URL location on tab change.
+ */
+const useRoutedTabs = (tabPaths, defaultTabIndex = 0) => {
+  const [tabIndex, setTabIndex] = useState(defaultTabIndex)
+  const { pathname } = useLocation()
+  const history = useHistory()
+
+  // Set the initial tabIndex from the URL on page load
+  useEffect(() => {
+    const matchedIndex = tabPaths.indexOf(pathname)
+    // eslint-disable-next-line no-magic-numbers
+    if (matchedIndex !== -1) {
+      setTabIndex(matchedIndex)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What was the reason for disabling the linter here vs putting the hook dependencies?

Reply:
The reason is because we only want to set the initial tab index from the location the first time. On subsequent tab changes, the tab index gets set in handleTabChange.

Reply:
Great question though. I tried to think about whether or not this usage is correct. Thoughts? Also @jackieo5023 @SirajChokshi

Reply:
I think using [this](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state) instead of the useEffect might be better :)

Reply:
^ this worked great 😄 thanks Jackie!